opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py

```diff
@@ -54,6 +54,22 @@
 
 _DEFAULT_OTEL_ATTRIBUTE_COUNT_LIMIT = 128
 _ENV_VALUE_UNSET = ""
+_EXCLUDED_ATTRIBUTES = {
+    # pseudo-private log-record attributes, they just get dropped
+    "args",
+    "msg",
+    "message",
+    "stack_info",
+    "exc_info",
+    "exc_text",
+    # attributes that are retained, but with a different name
+    # following semantic conventions
+    "pathname",
+    "funcName",
+    "lineno",
+    "thread",
+    "threadName",
```

Comment:
`taskName` was recently added to `_RESERVED_ATTRS` (now `_EXCLUDED_ATTRIBUTES`) to fix an endless recursion in python 3.12 so it also should probably be part of the `_EXCLUDED_ATTRIBUTES`, see: https://github.com/open-telemetry/opentelemetry-python/pull/3557

basically all attributes collected in the `_get_attributes` below will be validated (value types, valid key, ...) when creating OTel `LogRecord` (more specifically when the the [BoundedAttributes map](https://github.com/open-telemetry/opentelemetry-python/blob/v1.22.0/opentelemetry-api/src/opentelemetry/attributes/__init__.py#L100) is created). Invalid attributes will log a warning message which in turn will create another OTel `LogRecord` ...

the `taskName` introduced in python 3.12 is usually only set when logging is called from an asyncio task otherwise it is `None` and will set of the endless recursion (if log levels are enabled accordingly).

see also https://github.com/open-telemetry/opentelemetry-python/issues/3578

Reply:
I just ran the test suite with python3.12 without issues, as far as I can tell it's not present outside of an async context. That being said, I did have a `None` check in an old revision of this code, because the standard-lib `LoggingHandler` likes to un-set `exc_text` after formatting (it's even documented https://docs.python.org/3/library/logging.html#logging.Formatter.format), so before I decided that that field is just a cache that shouldn't be forwarded anyways I had this:
```
        attributes = {
            k: v
            for k, v in vars(record).items()
            if k not in _EXCLUDED_ATTRIBUTES and v is not None
        }
```
I'd like this better, because `taskName` sounds like valuable info if set.

Reply:
done